xray/plot/plot.py

```diff
@@ -452,6 +452,8 @@ def newplotfunc(darray, ax=None, xincrease=None, yincrease=None,
             # pcolormesh
             kwargs['extend'] = cmap_params['extend']
             kwargs['levels'] = cmap_params['levels']
+            if 'colors' in kwargs:
```

Comment:
Is compatibility with matplotlib conventions considered important? If so, you probably don't want to raise an error if contour is called with colors but without levels. That is a pretty common usage pattern when e.g. you want to make a black and white contour plot with auto-determined contour levels. I would just do ```plt.contour(field, colors='k')``` and let matplotlib pick the levels.